roles/ceph-client/tasks/create_users_keys.yml

```diff
@@ -64,19 +74,17 @@
 
 - name: list existing pool(s)
   command: >
-    {{ docker_exec_client_cmd | default('') }} ceph --cluster {{ cluster }}
+    {{ docker_exec_cmd | default('') }} ceph --cluster {{ cluster }}
     osd pool get {{ item.name }} size
   with_items: "{{ pools }}"
   register: created_pools
   run_once: true
   failed_when: false
-  delegate_to: "{{ groups.get(client_group_name)[0] }}"
-  when:
-    - copy_admin_key
+  delegate_to: "{{ delegated_node }}"
```

Comment:
We can't assume the client node has the admin_key is mon_group is not > 0 so we need `copy_admin_key` as a condition. Right?